doc/dynamic_profiles.md

```diff
@@ -0,0 +1,109 @@
+## Dynamic Profiles
+
+Somemtimes having static profile is not enough when it should be applied to wide range
+of machines and some dynamic content is useful. Currently autoyast supports
+three different ways how to achieve it. The first and the least flexible is 
+rules and classes [ TODO link to its documentation ]. The second way is creating
+profile via pre script [ TODO link to its documentation ]. It allows to use wide 
+range of languages and create profile from scratch or copy existing one and
+just modify it. Example of creating profile from scratch using shell is
+below. The third way is templates. Autoyast from SLE15 SP3 supports ERB templates
+[ TOO link to ERB documentation ]. ERB is default templating system in ruby
+and basically do preprocessing of specific directives with full power of ruby
+programming language. Example of such template is below. To work it properly it has
+to have suffix `.erb`.
```

Comment:
Mention hardware helper and also how to see its values from CLI: `ruby -r pp -r yast -r autoinstall/y2erb -e 'pp Y2Autoinstallation::Y2ERB::TemplateEnvironment.new.hardware'`